src/ensembl/src/content/app/custom-download/containers/content/result-holder/resultHolderHelper.ts

```diff
@@ -1,81 +1,96 @@
-  let endpoint = config.genesearchAPIEndpoint + '/genes/query?query=';
-
-  let endpointFields = '';
-  attributes.forEach((attribute: any) => {
-    endpointFields += attribute[2] + ',';
+import { keys, mapKeys, set, get, trim } from 'lodash';
+import JSONValue from 'src/shared/types/JSON';
+
+export const getProcessedAttributes = (flatSelectedAttributes: JSONValue) => {
+  const filteredAttributes = keys(flatSelectedAttributes).filter(
+    (key) => flatSelectedAttributes[key]
+  );
+  return filteredAttributes.map((value: string) => {
+    return value
+      .split('.default.')
+      .join('.')
+      .split('genes.')
+      .join('');
   });
+};
 
-  const endpointFilters: any = {
-    genome: 'homo_sapiens'
-  };
+export const getProcessedFilters = (filters: JSONValue) => {
+  const flatSelectedFilters: { [key: string]: boolean } = flattenObject(
+    filters
+  );
+
+  const selectedFilters = mapKeys(
+    flatSelectedFilters,
+    (value: boolean, key: string) => {
+      return key
+        .split('.default.')
+        .join('.')
+        .split('.genes.')
+        .join('.');
+    }
+  );
 
-  endpoint =
-    endpoint + JSON.stringify(endpointFilters) + '&fields=' + endpointFields;
+  const processedFilters = {};
 
-  return endpoint;
+  keys(selectedFilters).forEach((path) => {
+    set(processedFilters, path, selectedFilters[path]);
+  });
+  return processedFilters;
 };
 
-export const getSelectedAttributes = (attributes: any) => {
-  const selectedAttributes: any = [];
-
-  Object.keys(attributes).forEach((section) => {
-    Object.keys(attributes[section]).forEach((subSection) => {
-      Object.keys(attributes[section][subSection]).forEach((attributeId) => {
-        if (attributes[section][subSection][attributeId].isChecked === true) {
-          selectedAttributes.push([
-            section,
-            subSection,
-            attributes[section][subSection][attributeId].id,
-            attributes[section][subSection][attributeId].label
-          ]);
-        }
-      });
-    });
-  });
+export const getEndpointUrl = (
+  flatSelectedAttributes: JSONValue,
+  selectedFilters: JSONValue,
+  method: string = 'query'
+) => {
+  const processedAttributes = getProcessedAttributes(flatSelectedAttributes);
+  const processedFilters = getProcessedFilters(selectedFilters);
 
-  return selectedAttributes;
-};
+  let endpoint = config.genesearchAPIEndpoint + `/genes/${method}?query=`;
+
+  const endpointFilters: JSONValue = {
+    genome: 'homo_sapiens'
+  };
 
-export const getSelectedFilters = (filters: any) => {
-  const selectedFilters: any = {};
+  // FIXME: Temporarily apply the filters locally
+  const gene_ids = get(processedFilters, 'genes.limit_to_genes');
+  const gene_biotypes = get(processedFilters, 'genes.biotype');
+  const gene_source = get(processedFilters, 'genes.gene_source');
+
+  if (gene_ids) {
+    endpointFilters.id = gene_ids
+      .join(',')
+      .split(',')
```

Comment:
what does `.join(',').split(',')` do?

Reply:
It's doing this:
['1,2,3', '4,5,6'] => [1,2,3,4,5,6]